src/include/txn.i

```diff
@@ -819,6 +840,10 @@ __wt_txn_read(WT_SESSION_IMPL *session, WT_CURSOR_BTREE *cbt, WT_UPDATE *upd, WT
 
     /* There is no BIRTHMARK in the history store file. */
     WT_ASSERT(session, upd == NULL || upd->type != WT_UPDATE_BIRTHMARK);
+
+    if (upd == NULL)
```

Comment:
Explicitly return tombstones when we get an absence of a value. This should stop the surrounding code from falling back to the on-disk value. That code should get ripped at some point but let's just aim for functional correctness first.